### PR TITLE
Adding a new page on building adaptive apps

### PR DESCRIPTION
**Pull request by sfshaza2**
Staged: https://sz-flutter-2.firebaseapp.com/docs/development/ui/layout/building-adaptive-apps

cc @RedBrogdon

### CODE REVIEW

**Comment by esDotDev** [comment]
This LGTM, thanks for all the help Shams!

File: src/docs/development/ui/layout/building-adaptive-apps.md [hunk 1]
```diff
@@ -0,0 +1,1217 @@
...
+
+`MouseRegion` is also useful for creating custom
+rollover and hover effects:
+
+<!--skip-->
+```dart
+return MouseRegion(
+ onEnter: (_) => setState(() => _isMouseOver = true),
+ onExit: (_) => setState(() => _isMouseOver = false),
+ onHover: (e) => print((e as PointerHoverEvent).localPosition);
```
Comment (esDotDev) [inline src/docs/development/ui/layout/building-adaptive-apps.md:816]: This line would be better written as:
`onHover: (PointerHoverEvent e) => print(e.localPosition);`

File: src/docs/development/ui/layout/building-adaptive-apps.md [hunk 2]
```diff
@@ -0,0 +1,1217 @@
...
+// Define a class for each type of shortcut action you want
+class CreateNewItemIntent extends Intent {
+ const CreateNewItemIntent();
+}
+
+Widget build(BuildContext context) {
+ return Shortcuts(
+   // Bind intents to key combinations
+   shortcuts: <LogicalKeySet, Intent>{
+     LogicalKeySet(LogicalKeyboardKey.control, LogicalKeyboardKey.keyN): CreateNewItemIntent(),
```
Comment (esDotDev) [inline src/docs/development/ui/layout/building-adaptive-apps.md:701]: I've just been informed this is legacy method, new method is:
```
shortcuts: <ShortcutActivator, Intent>{
  SingleActivator(LogicalKeyboardKey.keyN, control: true): CreateNewItemIntent(),
```

**Review by esDotDev — CHANGES_REQUESTED** [review]

File: src/docs/development/ui/layout/building-adaptive-apps.md (hunk 1 above)
Comment (dkwingsmt) [reply src/docs/development/ui/layout/building-adaptive-apps.md:816]: `onHover: (PointerHoverEvent e) => print(e.localPosition),`
(the last comma)

**Comment by kevmoo** [comment]
@Sfshaza – are we going to update https://flutter.dev/go/adaptive to point here?

**Pull request merged**